src/sched/nop_asm_impl.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#define _ASM_NOP asm volatile("nop")
+
+// Cannot use `do { } while (0)` here because there would be huge differences between -O0 and -O1.
+#define _2X(statement) \
+    {                  \
+        statement;     \
+        statement;     \
+    }
+
+#define _2X_NOP _2X(_ASM_NOP)
+#define _4X_NOP _2X(_2X_NOP)
+#define _8X_NOP _2X(_4X_NOP)
+#define _16X_NOP _2X(_8X_NOP)
+#define _32X_NOP _2X(_16X_NOP)
+
+// Cannot use `do { } while (0)` here because there would be huge differences between -O0 and -O1.
```

Comment:
I only said that for a single nop, 1us should be fine.

Reply:
What do you mean? We have done experiments for it.

Reply:
You should be able to use do-while here

Reply:
I only mean you cannot use do-while around each nop, because it cannot be optimized out by default. But run it only once is fine.